### PR TITLE
Deprecate marching_cubes_classic

### PR DESCRIPTION
**Pull request by rfezzani**
## Description

What about deprecating `marching_cubes_classic`?

I propose in this PR to create a unique `marching_cubes` function and to activate two deprecation cycles:
- `marching_cubes_classic` in favor of `marching_cubes` with the option `use_classic=2`
- `marching_cubes_lewiner` in favor of `marching_cubes`

During the deprecation cycle, `marching_cube` acts as a wrapper for the two other implementations, and at the end of the deprecation cycle, `marching_cubes_classic` is removed and `marching_cubes_lewiner` is renamed `marching_cubes`.

## Checklist

- [x] [Docstrings for all functions](https://github.com/numpy/numpy/blob/master/doc/example.py)
- [x] Gallery example in `./doc/examples` (new features only)
- [ ] Benchmark in `./benchmarks`, if your changes aren't covered by an
  existing benchmark
- [x] Unit tests
- [x] Clean style in [the spirit of PEP8](https://www.python.org/dev/peps/pep-0008/)

## For reviewers

- [ ] Check that the PR title is short, concise, and will make sense 1 year
  later.
- [ ] Check that new functions are imported in corresponding `__init__.py`.
- [ ] Check that new features, API changes, and deprecations are mentioned in
      `doc/release/release_dev.rst`.
- [ ] Consider backporting the PR with `@meeseeksdev backport to v0.14.x`

### CODE REVIEW

File: skimage/measure/_marching_cubes_lewiner.py [hunk 1]
```diff
@@ -1,10 +1,139 @@
...
 import base64
 
 import numpy as np
 
 from . import _marching_cubes_lewiner_luts as mcluts
 from . import _marching_cubes_lewiner_cy
+from ._marching_cubes_classic import _marching_cubes_classic
+
+
+def marching_cubes(volume, level=None, spacing=(1., 1., 1.),
```
Comment (jni) [inline skimage/measure/_marching_cubes_lewiner.py]: Since this is a new function, let's make arguments after `level` keyword-only

File: skimage/measure/_marching_cubes_lewiner.py [hunk 2]
```diff
@@ -1,10 +1,139 @@
...
+                                       allow_degenerate, use_classic=False)
+    elif use_classic == 1:
+        return _marching_cubes_lewiner(volume, level, spacing,
+                                       gradient_direction, step_size,
+                                       allow_degenerate, use_classic=True)
+    elif use_classic == 2:
+        return _marching_cubes_classic(volume, level, spacing,
+                                       gradient_direction)
+    else:
+        raise ValueError("use_classic should be 0, 1 or 2.")
```
Comment (jni) [inline skimage/measure/_marching_cubes_lewiner.py]: I suggest `method=`, with some string or enum choices, and deprecating `use_classic`. Using an Int flag is very unpythonic.

File: skimage/measure/_marching_cubes_lewiner.py [hunk 3]
```diff
@@ -108,16 +237,35 @@ def marching_cubes_lewiner(volume, level=None, spacing=(1., 1., 1.),
...
-    skimage.measure.marching_cubes_classic
+    skimage.measure.marching_cubes
     skimage.measure.mesh_surface_area
+
+    """
+
+    # Deprecate the function in favor of marching_cubes
+    warnings.warn("marching_cubes_lewiner is deprecated in favor of "
+                  + "marching_cubes. marching_cubes_lewiner will "
+                  + "be removed in version 0.19",
```
Comment (jni) [inline skimage/measure/_marching_cubes_lewiner.py]: No need for `+`, as Python will implicitly continue strings.

File: skimage/measure/_marching_cubes_lewiner.py [hunk 4]
```diff
@@ -194,26 +350,56 @@ def _get_mc_luts():
...
-                _to_array(mcluts.TILING12_1_1), _to_array(mcluts.TILING12_1_1_), _to_array(mcluts.TILING12_1_2),
-                _to_array(mcluts.TILING12_2), _to_array(mcluts.TILING12_2_), _to_array(mcluts.TILING13_1),
-                _to_array(mcluts.TILING13_1_), _to_array(mcluts.TILING13_2), _to_array(mcluts.TILING13_2_),
-                _to_array(mcluts.TILING13_3), _to_array(mcluts.TILING13_3_), _to_array(mcluts.TILING13_4),
-                _to_array(mcluts.TILING13_5_1), _to_array(mcluts.TILING13_5_2), _to_array(mcluts.TILING14),
-
-                _to_array(mcluts.TEST3), _to_array(mcluts.TEST4), _to_array(mcluts.TEST6),
-                _to_array(mcluts.TEST7), _to_array(mcluts.TEST10), _to_array(mcluts.TEST12),
-                _to_array(mcluts.TEST13), _to_array(mcluts.SUBCONFIG13),
-                )
```
Comment (jni) [inline skimage/measure/_marching_cubes_lewiner.py]: Please revert these changes, and anything that is just "PEP8" for existing code, as style changes like these make it hard to track code history.

**Comment by jni** [comment]
@rfezzani in principle I'm very happy with this approach! I noticed it recently because I've been working a bit more with surface data — the name `marching_cubes_lewiner` didn't really gel with me!

I've left some minor comments that should be addressed before merging, but mostly I think I agree with the change as is...! Thank you!

File: skimage/measure/_marching_cubes_lewiner.py (hunk 4 above)
Comment (rfezzani) [reply skimage/measure/_marching_cubes_lewiner.py]: I just fixed pep8 because it was making CI fail... :-)

File: skimage/measure/_marching_cubes_lewiner.py (hunk 2 above)
Comment (rfezzani) [reply skimage/measure/_marching_cubes_lewiner.py]: What about `method=` one of `lewiner`, `lorensen` and `_lorensen`?

**Comment by rfezzani** [comment]
Hi @jni, thank you for your review. I addressed your comments... **pep8speaks** is complaining again :smile:

**Comment by Leguark** [comment]
Does it make sense adding to this pull request the masked marching cubes #3829 ? I am afraid that after this refactorization, merging the masked marching cube is not going to be trivial

**Comment by rfezzani** [comment]
Hi @Leguark, thank you for your contribution ;-). I don't think that core team will agree with merging both PR in one, and I feel that the merging will not be that complicated, but well... Let @jni decide.

BTW, I made a small review to #3829.

**Comment by emmanuelle** [comment]
Thank you @rfezzani ! merging !

**Comment by rfezzani** [comment]
You welcome @emmanuelle!
